application/recognition/image_recognition.py

```diff
@@ -0,0 +1,48 @@
+"""Image Recognition"""
+from eventlet import spawn_n
+from requests import post
+from application.helpers.artifact_helper import build_url
+from flask import current_app
+
+
+def recognize_image(file_url, object_id):
+    """Called by classes """
+    spawn_n(_work_asynchronously, file_url, object_id)
+
+
+def _call_google_api(url):
+    """Does API call to could vision api"""
+    api_url = current_app.config.get('CLOUD_VISION_API_URL')
+    api_key = current_app.config.get('CLOUD_VISION_API_KEY')
+    r = post(url=api_url,
+             data={
+                 "requests": [
+                     {
+                         "features": [
+                             {"type": "LABEL_DETECTION"},
+                             {"type": "TEXT_DETECTION"}
+                         ],
+                         "image": {
+                             "source": {"imageUri": url}
+                         }
+                     }
+                 ]
+             },
+             params={'key': api_key})
+    return r.json()
+
+
+def _work_asynchronously(file_url, artifact):
+    """Private method called asynchronously for image recognition."""
+    image_url = build_url(file_url)
```

Comment:
you could also probably just add a method to artifact: `artifact.current_url()` or something similar.

Reply:
This is now a static method in artifact schema.